pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -281,19 +283,149 @@ def __call__(self, time: float):
+                testing_type_general_pop = testing_params['sympt_pcr'][2]
+                r = random.random()
+                type_r = random.random()
+
+                if (person.is_symptomatic() and
+                   person.date_positive is None):
+                    if person.care_home_resident:
+                        test_probability = testing_prob_care_home
+                        type_probability = testing_type_care_home
+                    elif person.key_worker:
+                        test_probability = testing_prob_key_worker
+                        type_probability = testing_type_key_worker
+                    else:
+                        test_probability = testing_prob_general_pop
+                        type_probability = testing_type_general_pop
+
+                    if r < test_probability:
+                        if type_r < type_probability:
+                            cell.enqueue_PCR_testing(person)
+                        else:
+                            cell.enqueue_LFT_testing(person)
+
+                if (person.date_positive is not None and
+                    (person.next_infection_status in
+                     [InfectionStatus.Dead, InfectionStatus.Recovered])):
+                    person.date_positive = None
+
+    def asympt_uninf_testing_queue(self, person_list: list, time):
+        """ Adds asymptomatic and uninfected people to a testing queue
+        with a given probability depedent on their status as either a care
+        home resident or key worker.
+
+        Detailed description of the implementation can be found in github wiki:
+        https://github.com/SABS-R3-Epidemiology/epiabm/wiki/Interventions#testing
+
+        Parameters
+        ----------
+        person_list : list
+            list of (cell, person) tuples giving the list of people
+            to be added to a testing queue and their cell.
+        time : float
+            current time point to determine whether uninfected indivuals
+            should stop being considered as positive.
+
+        """
+        if hasattr(Parameters.instance(), 'intervention_params'):
+            if 'testing' in Parameters.instance().\
+               intervention_params.keys():
+                testing_params = Parameters.instance().\
+                    intervention_params['testing']
+                testing_prob_care_home = testing_params[
+                    'testing_asympt_uninf'][0]
+                testing_prob_key_worker = testing_params[
+                    'testing_asympt_uninf'][1]
+                testing_prob_general_pop = testing_params[
+                    'testing_asympt_uninf'][2]
+                testing_type_care_home = testing_params[
+                    'asympt_uninf_pcr'][0]
+                testing_type_key_worker = testing_params[
+                    'asympt_uninf_pcr'][1]
+                testing_type_general_pop = testing_params[
+                    'asympt_uninf_pcr'][2]
```

Comment:
Do we need to unpack all of these just to reassign them a few lines later (L413 onwards) could we just read them in there